### ISSUE TITLE
How to right pad data in Tensorflow Transform to feed into a LSTM ?

### ISSUE THREAD

**Issue by anantvir** [tensorflow/tensorflow#38610]
I am building a data pipeline using Tensorflow Transform using Apache Beam. Inside the preprocessing function I am generating vocabulary which is converting my input sentences to list of integers.
My data has 3 features:
Column 1 = Context (dtype = String) (Sentences of varying Length)
Column 2 = Utterance (dtype = String) (Sentences of varying Length)
Column 3 = Label (0/1)
I want to right pad my list of integers as soon as they are generated in the preprocessing function below to a max length of 160 words.
Example:
"I love Pizza" --> [34, 67, 78] --> Max length I want = 10
then I want [34, 67, 78,0,0,0,0,0,0,0] and if the length of my sentence is already greater than 10, then I want to trim the extra portion to make it length  =10

Now to use tf.keras.preprocessing.sequence.pad_sequences you need as input a list of sequences but as shown below, my mapped_context and mapped_utterance are tf.int64 tensors. So I am not able to use the padding functionality of Keras.
Can someone please help me achieve this ?

The reference code I am following is Tensorflow Sentiment Analysis Example:
https://github.com/tensorflow/transform/blob/599691c8b94bbd6ee7f67c11542e7fef1792a566/examples/sentiment_example.py
------------------------------------------------------------------------------------------------------------

My code's preprocessing function is below:

![image](https://user-images.githubusercontent.com/27782859/79493187-1a195300-7fef-11ea-9d5c-560787e137b8.png)

My Dataset is below:

![image](https://user-images.githubusercontent.com/27782859/79533358-a5bdce80-8045-11ea-8e40-d820cf935f42.png)

**Comment by khimraj** [comment]
@anantvir, you can right pad a sequence using following code.
```
from keras.preprocessing.sequence import pad_sequences
padded = pad_sequences(sequences, padding='post', maxlen=10, truncating='post')
```

**Comment by anantvir** [comment]
> 
> 
> @anantvir, you can right pad a sequence using following code.
> 
> ```
> from keras.preprocessing.sequence import pad_sequences
> padded = pad_sequences(sequences, padding='post', maxlen=10, truncating='post')
> ```

That was my question. To use pad_sequences, the sequences argument should be a list of sequences
example
[
  [83, 91, 1, 645, 1253, 927],
  [73, 8, 3215, 55, 927],
  [71, 1331, 4231]
]
In my case I do not have this list. What I have is a tensor. Because it is inside the preprocessing_fn in Apache Beam pipeline. Please see the code screenshot attached.

**Comment by anantvir** [comment]
This is my dataset. I want to feed both "Context" and "Utterance" column into a LSTM.

![image](https://user-images.githubusercontent.com/27782859/79532773-24b20780-8044-11ea-8a28-e113ca9183cd.png)

**Comment by anantvir** [comment]
can someone please update me on this issue ?  @ravikyram @khimraj ?

**Comment by khimraj** [comment]
@anantvir, You need to first use Tokenizer to convert text into sequences. For more detail refer [this](https://towardsdatascience.com/multi-class-text-classification-with-lstm-using-tensorflow-2-0-d88627c10a35) tutorial.

**Comment by anantvir** [comment]
> 
> 
> @anantvir, You need to first use Tokenizer to convert text into sequences. For more detail refer [this](https://towardsdatascience.com/multi-class-text-classification-with-lstm-using-tensorflow-2-0-d88627c10a35) tutorial.

@khimraj  Are you familiar with Tensorflow Transform ? I am sking this because as I explained above, I want to make all sentences of equal length inside the preprocessing_fn in the apache beam pipeline. We cannot use Tokenizer as it also requires a list of sequences. That is WHAT I DONT HAVE. I want to pad tensors not sequences.

Please refer to: 
https://github.com/tensorflow/transform/blob/599691c8b94bbd6ee7f67c11542e7fef1792a566/examples/sentiment_example.py

This is the code I am using as a reference and I want to make all sentences of equal length inside the preprocessing_fn starting at line 181.

Please let me know in case you need any further information.

**Comment by khimraj** [comment]
@anantvir, Sorry for misunderstanding your query. I will try to solve problem after reading your references.

**Comment by anantvir** [comment]
Any updates ?

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
Thanks!
